packages/admin-panel/src/VizBuilderApp/components/PreviewSection.js

```diff
@@ -0,0 +1,150 @@
+import MuiTab from '@material-ui/core/Tab';
+import MuiTabs from '@material-ui/core/Tabs';
+import { Chart, Table } from '@tupaia/ui-components/lib/chart';
+import { FlexSpaceBetween, FetchLoader } from '@tupaia/ui-components';
+import { TabPanel } from './TabPanel';
+import { useStore } from '../store';
+import { useReportPreview } from '../api';
+import { JsonEditor } from './JsonEditor';
+import { IdleMessage } from './IdleMessage';
+
+const PreviewTabs = styled(MuiTabs)`
+  background: white;
+  border: 1px solid ${({ theme }) => theme.palette.grey['400']};
+  border-bottom: none;
+
+  .MuiTabs-indicator {
+    height: 5px;
+  }
+`;
+
+const PreviewTab = styled(MuiTab)`
+  font-size: 15px;
+  line-height: 140%;
+  font-weight: 400;
+  min-width: 100px;
+  padding-top: 20px;
+  padding-bottom: 20px;
+`;
+
+const PanelTabPanel = styled.div`
+  flex: 1;
+  background: white;
+  border: 1px solid ${({ theme }) => theme.palette.grey['400']};
+  border-top: none;
+`;
+
+const StyledTable = styled(Table)`
+  table {
+    border-top: 1px solid ${({ theme }) => theme.palette.grey['400']};
+    border-bottom: 1px solid ${({ theme }) => theme.palette.grey['400']};
+  }
+`;
+
+const Container = styled(FlexSpaceBetween)`
+  align-items: stretch;
+  flex-wrap: wrap;
+  height: 100%;
+`;
+
+const TableContainer = styled.div`
+  display: flex;
+  height: 100%;
+`;
+
+const ChartContainer = styled.div`
+  display: flex;
+  padding: 4rem 2rem 2rem;
+  max-height: 550px;
+  min-height: 450px;
+  min-width: 540px;
+  flex: 2;
+  border-top: 1px solid ${({ theme }) => theme.palette.grey['400']};
+`;
+
+const EditorContainer = styled.div`
+  min-width: 440px;
+  min-height: 500px;
+  flex: 1;
+  border-top: 1px solid ${({ theme }) => theme.palette.grey['400']};
+  border-left: 1px solid ${({ theme }) => theme.palette.grey['400']};
+
+  > div {
+    width: 100%;
+    height: 100%;
+  }
+
+  .jsoneditor {
+    border: none;
+  }
+`;
+
+export const PreviewSection = ({ enabled, setEnabled }) => {
+  const [state, { setPresentation }] = useStore();
+  const { data = [], isIdle, isLoading, isFetching, isError, error } = useReportPreview(
+    state,
+    enabled,
+    setEnabled,
+  );
+  const [tab, setTab] = useState(0);
+
+  const handleChange = (event, newValue) => {
+    setTab(newValue);
+  };
+
+  const viewContent = { data, ...state.presentation };
+
+  return (
+    <>
+      <PreviewTabs
+        value={tab}
+        variant="fullWidth"
+        indicatorColor="primary"
+        textColor="primary"
+        onChange={handleChange}
+      >
+        <PreviewTab label="Data Preview" />
+        <PreviewTab label="Chart Preview" />
+      </PreviewTabs>
+      <TabPanel isSelected={tab === 0} Panel={PanelTabPanel}>
+        <TableContainer>
+          {isIdle ? (
+            <IdleMessage />
+          ) : (
+            <FetchLoader isLoading={isLoading || isFetching} isError={isError} error={error}>
+              <StyledTable viewContent={viewContent} />
```

Comment:
I'm just using the table flippa table at the moment to preview the data. I'm not 100% sure if this will be what Rohan is expecting so I will keep the data table card in progress for changes.